### PR TITLE
Complete new nav

### PR DESCRIPTION
**Pull request by Hidde-Heijnen**
completed @BenjaminMeijs Nav dropdown and mobile nav

### CODE REVIEW

File: src/components/layout/Navbar.tsx [hunk 1]
```diff
@@ -42,29 +60,50 @@ const navItems: NavItem[] = [
...
   {
     label: 'SearchSECO',
     pages: [
-      { label: 'Query', url: '/query' },
-      { label: 'Mining', url: '/mining' },
+      {
+        label: 'Query',
+        url: '/query',
+        icon: HiOutlineDocumentMagnifyingGlass,
+        description: 'Query the SearchSECO database using your $SECOINS',
```
Comment (ThijnK) [inline src/components/layout/Navbar.tsx:74]: Please use the `TOKENS` variable for the symbol of the SECOIN.
Also, we should decide on a consistent way to either always use $ in front of the symbol, or not (currently we do not, but can change if you so desire)

File: src/components/layout/Navbar.tsx [hunk 2]
```diff
@@ -42,29 +60,50 @@ const navItems: NavItem[] = [
...
+      },
+      {
+        label: 'Mining',
+        url: '/mining',
+        icon: HiOutlineTerminal,
+        description: 'Claim your mining rewards in $SECOINS or $SECOREP',
```
Comment (ThijnK) [inline src/components/layout/Navbar.tsx:80]: Same here as above comment for the SECOIN and SECOREP, use `TOKENS.secoin.symbol` and `TOKENS.rep.symbol`

**Review by ThijnK — CHANGES_REQUESTED** [review]
Few minor changes, see comments

File: src/components/layout/Navbar.tsx [hunk 3]
```diff
@@ -42,29 +61,50 @@ const navItems: NavItem[] = [
...
   {
     label: 'SearchSECO',
     pages: [
-      { label: 'Query', url: '/query' },
-      { label: 'Mining', url: '/mining' },
+      {
+        label: 'Query',
+        url: '/query',
+        icon: HiOutlineDocumentMagnifyingGlass,
+        description: `Query the SearchSECO database using your ${TOKENS.secoin.symbol} tokens.}`,
```
Comment (ThijnK) [inline src/components/layout/Navbar.tsx:75]: There's a period at the end of this description, while there isn't one in the description of Mining page.
You know what they say, it's all in the details

File: src/components/ui/Sheet.tsx [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+ */
+
+/**
+ * @fileoverview This file contains the Sheet component, which is a wrapper around the Radix UI react dialog.
+ * Extends the Dialog component to display content that complements the main content of the screen.
+ * @see https://www.radix-ui.com/docs/primitives/components/dialog - Radix UI Dialog Primitive
+ * inspired by https://ui.shadcn.com/docs/primitives/sheet
+ */
+
+'use client';
```
Comment (ThijnK) [inline src/components/ui/Sheet.tsx:16]: I know you love Next.js, but alas, we're not using it, so no need for this

File: src/components/layout/Navbar.tsx [hunk 5]
```diff
@@ -42,29 +61,50 @@ const navItems: NavItem[] = [
...
     ],
-  },
-  {
-    label: 'Settings',
-    url: '/settings',
   },
 ];
 
-const Navitempage = ({ item }: { item: NavItemPage }) => {
+const Navitempage = ({
```
Comment (ThijnK) [inline src/components/layout/Navbar.tsx:91]: What happened to naming convention for components??
Same for the other components in this file

File: src/components/layout/Navbar.tsx [hunk 6]
```diff
@@ -150,12 +186,117 @@ const Navbar = () => {
...
+            </NavLink>
+          ))}
+      </div>
+    </div>
+  );
+};
+
 export default Navbar;
+
+function MobileNav() {
```
Comment (ThijnK) [inline src/components/layout/Navbar.tsx:266]: Why is this suddenly not an arrow function, while all other components are?

File: src/components/layout/Navbar.tsx [hunk 7]
```diff
@@ -73,74 +113,70 @@ const Navitempage = ({ item }: { item: NavItemPage }) => {
...
-    return <Navitemcollection item={item as NavItemCollection} />;
+    return (
+      <>
+        <Navitemcollection
+          item={item as NavItemCollection}
+          className={className + ' hidden lg:flex'}
+        />
+        <div className="w-full rounded-3xl border border-border text-sm leading-5 shadow-lg md:col-span-2 lg:hidden">
+          <DropdownContent
+            title={'SearchSECO'}
```
Comment (ThijnK) [inline src/components/layout/Navbar.tsx:164]: Accidental hard coded value?

**Review by ThijnK — CHANGES_REQUESTED** [review]
See comments above

**Comment by Hidde-Heijnen** [comment]
> See comments above

Will fix today, after merge to main + fix env on vercel?

File: src/components/layout/Navbar.tsx (hunk 5 above)
Comment (Hidde-Heijnen) [reply src/components/layout/Navbar.tsx:91]: Dat was @BenjaminMeijs whoops, anders had de naam conflict met de types

**Comment by Hidde-Heijnen** [comment]
Check now @ThijnK

**Review by ThijnK — COMMENTED** [review]
LGTM!

**Pull request merged**